ts/Series/ErrorBar/ErrorBarSeries.ts

```diff
@@ -0,0 +1,276 @@
+         * @default #000000
+         * @since   3.0
+         * @product highcharts
+         */
+        color: '${palette.neutralColor100}',
+
+        grouping: false,
+
+        /**
+         * The parent series of the error bar. The default value links it to
+         * the previous series. Otherwise, use the id of the parent series.
+         *
+         * @since   3.0
+         * @product highcharts
+         */
+        linkedTo: ':previous',
+
+        tooltip: {
+            pointFormat: '<span style="color:{point.color}">\u25CF</span> {series.name}: <b>{point.low}</b> - <b>{point.high}</b><br/>'
+        },
+
+        /**
+         * The line width of the whiskers, the horizontal lines marking
+         * low and high values. When `null`, the general
+         * [lineWidth](#plotOptions.errorbar.lineWidth) applies.
+         *
+         * @sample {highcharts} highcharts/plotoptions/error-bar-styling/
+         *         Error bar styling
+         *
+         * @type    {number}
+         * @since   3.0
+         * @product highcharts
+         */
+        whiskerWidth: null as any
+
+    });
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<ErrorBarPoint> = void 0 as any;
+    public options: ErrorBarSeriesOptions = void 0 as any;
+    public points: Array<ErrorBarPoint> = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    // Get the width and X offset, either on top of the linked series
+    // column or standalone
+    public getColumnMetrics(): ColumnMetricsObject {
+        return (
+            (this.linkedParent && this.linkedParent.columnMetrics) ||
+            ColumnSeries.prototype.getColumnMetrics.call(this)
+        );
+    }
+
+    public drawDataLabels(): void {
+        var valKey = this.pointValKey;
+
+        if (AreaRangeSeries) {
+            AreaRangeSeries.prototype.drawDataLabels.call(this);
+            // Arearange drawDataLabels does not reset point.y to high,
+            // but to low after drawing (#4133)
+            this.data.forEach(function (point: ErrorBarPoint): void {
+                point.y = (point as any)[valKey];
+            });
+        }
+    }
+
+    // return a plain array for speedy calculation
+    public toYData(point: ErrorBarPoint): Array<number> {
+        return [point.low, point.high];
+    }
+}
+
+
+/* *
+ *
+ *  Prototype properties
+ *
+ * */
+
+interface ErrorBarSeries extends BoxPlotSeries {
+    doQuartiles: boolean;
+    linkedParent: ErrorBarSeries;
+    pointArrayMap: Array<string>;
+    pointClass: typeof ErrorBarPoint;
+    pointValKey: string;
+}
+
+extend(ErrorBarSeries.prototype, {
+    // array point configs are mapped to this
+    pointArrayMap: ['low', 'high'],
+    pointValKey: 'high', // defines the top of the tracker
+    doQuartiles: false
+});
+
+
+/* *
+ *
+ *  Registry
+ *
+ * */
+
+BaseSeries.registerSeriesType('errorbar', ErrorBarSeries);
+
+declare module '../../Core/Series/SeriesType' {
```

Comment:
Switch statements, please. We might get more specific about the types of the registry function.